docs/releases/upgrading/upgrade-v10.0.md

```diff
@@ -0,0 +1,270 @@
+
+#### OpenSearch: prepare upgrade
+
+At this point, if you are planning to upgrade to OpenSearch v1 or v2, you will need to make a few changes
+before upgrading InvenioRDM. If you will keep Elasticsearch v7, please skip this step and go to [Upgrade InvenioRDM](#upgrade-inveniordm).
+
+Elasticsearch v7 and OpenSearch v1 are very similar and the upgrade is rather simple (see the official OpenSearch documentation [here](https://opensearch.org/docs/2.0/upgrade-to/upgrade-to/)). For OpenSearch v2, the only breaking change, affecting InvenioRDM, is the removal of `doc_type` support, more information [here](https://github.com/opensearch-project/OpenSearch/issues/2480).
+
+1. Prepare your new OpenSearch v1/v2 cluster. For local development, make sure that you also have upgraded OpenSearch in your `docker-services.yml`, as in [here](https://github.com/inveniosoftware/cookiecutter-invenio-rdm/blob/e8082e71453ca867e503149538c67c66eb182360/%7B%7Bcookiecutter.project_shortname%7D%7D/docker-services.yml).
+2. In your instance, edit the `Pipfile` and replace `invenio-app-rdm = {extras = [...  "elasticsearch7"]...` with `opensearch2` (or `opensearch2`). For example, in the `Pipfile`:
+    ```bash
+    [packages]
+    -invenio-app-rdm = {extras = ["postgresql", "elasticsearch7"], version = "~=10.0.0"}
+    +invenio-app-rdm = {extras = ["postgresql", "opensearch2"], version = "~=10.0.0"}
+    ```
+3. In your instance, edit the `.invenio` file:
+    ```bash
+    -search = elasticsearch7
+    +search = opensearch2
+    ```
+4. Do you have any customisation or custom module that contains Elasticsearch v7 mappings? If not, go to the next step.
+   If yes, make sure that you copy the `v7` folder and paste it in the same location, with the new name `os-v2` (or `os-v1`).
+   The final folder structure should look like this (do **not** forget the `__init__.py` files, they are required!):
+
+    ```bash
+    my-custom-module
+    |-- mappings
+        |-- __init__.py
+        |-- os-v2
+            |-- __init__.py
+            |-- <record>
+                |-- <record>-v1.0.0.json
+        |-- v7
+            |-- __init__.py
+            |-- <record>
+                |-- <record>-v1.0.0.json
+    ```
+
+    !!! warning "OpenSearch v2 breaking change"
+
+        Make sure that you have removed the `_type` field everywhere. OpenSearch v2 will fail if still present!
+
+5. Rename the `INVENIO_SEARCH_ELASTIC_HOSTS` configuration variable to `INVENIO_SEARCH_HOSTS` and change its value to point to the new cluster.
+6. Uninstall the Elasticsearch Python libraries:
+    ```bash
+    cd <my-site>
+    pipenv run pip uninstall -y elasticsearch elasticsearch-dsl
+    ```
+
+#### Upgrade InvenioRDM
+
+Bump the RDM version, create a folder for your custom fields and rebuild the assets:
+
+```bash
+cd <my-site>
+# Upgrade to InvenioRDM v10
+invenio-cli packages update 10.0.0
+```
+
+### Re-index data
```

Comment:
Im not sure if I understand correctly, why are the reindex (this and the code snippet) separated? Is the build assets that happens in the middle necessary for the reindex? Otherwise, can we put the reindex all together? I think this will be simlified if we put it in an annex (since we have to reference it 3 times.